src/onegov/org/views/resource.py

```diff
@@ -970,25 +971,36 @@ def handle_delete_resource(self: Resource, request: OrgRequest) -> None:
 
     request.assert_valid_csrf_token()
     tickets = TicketCollection(request.session)
+    payments = PaymentCollection(request.session)
 
-    def handle_reservation_tickets(reservation: BaseReservation) -> None:
-        ticket = tickets.by_handler_id(reservation.token.hex)
+    def handle_reservation_ticket(ticket: ReservationTicket) -> None:
         if ticket:
             assert request.current_user is not None
 
             close_ticket(ticket, request.current_user, request)
             ticket.create_snapshot(request)
 
-            payment = ticket.handler.payment
-            if (payment and PaymentCollection(request.session).query()
-                    .filter_by(id=payment.id).first()):
-                PaymentCollection(request.session).delete(payment)
+            if ticket.payment:
+                for invoice_item in ticket.invoice.items:
+                    invoice_item.payments = []
+                    request.session.delete(invoice_item)
+
+                if ticket.invoice:
+                    request.session.delete(ticket.invoice)
```

Comment:
Invoices can exist, even if there is no payment (if the total amount is ≤0), so this check needs to happen outside the `if ticket.payment` check.